iaf-aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/services/conv/impl/convertors/MailConvertor.java

```diff
@@ -0,0 +1,227 @@
+
+	private static final float MaxImageWidthInPoints = PageConvertUtil
+			.convertCmToPoints(PageConvertUtil.PAGE_WIDHT_IN_CM - 2 * 1.1f);
+	private static final float MaxImageHeightInPoints = PageConvertUtil
+			.convertCmToPoints(PageConvertUtil.PAGE_HEIGTH_IN_CM - 2 * 1.1f);
+
+	private CisConversionService cisConversionService;
+
+	// contains mapping from MediaType to the LoadOption for the aspose word
+	// conversion.
+	private static final Map<MediaType, com.aspose.email.LoadOptions> MEDIA_TYPE_LOAD_FORMAT_MAPPING;
+
+	static {
+		Map<MediaType, com.aspose.email.LoadOptions> map = new HashMap<>();
+		map.put(new MediaType("message", "rfc822"), null);
+		map.put(new MediaType("message", "rfc822.concept"), null);
+		map.put(new MediaType("message", "rfc822.ddcim"), null);
+		map.put(new MediaType("application", "vnd.ms-outlook"), null);
+		map.put(new MediaType("application", "vnd.ms-tnef"), new TnefLoadOptions());
+
+		MEDIA_TYPE_LOAD_FORMAT_MAPPING = Collections.unmodifiableMap(map);
+	}
+
+	MailConvertor(CisConversionService cisConversionService, String pdfOutputLocation) {
+		super(pdfOutputLocation,
+				MEDIA_TYPE_LOAD_FORMAT_MAPPING.keySet().toArray(new MediaType[MEDIA_TYPE_LOAD_FORMAT_MAPPING.size()]));
+
+		this.cisConversionService = cisConversionService;
+	}
+
+	@Override
+	public void convert(MediaType mediaType, File file, CisConversionResult result, ConversionOption conversionOption)
+			throws Exception {
+		MailMessage eml = null;
+
+		try (FileInputStream inputStream = new FileInputStream(file)) {
+			eml = MailMessage.load(inputStream, MEDIA_TYPE_LOAD_FORMAT_MAPPING.get(mediaType));
+
+			AttachmentCollection attachments = eml.getAttachments();
+
+			LOGGER.debug("cc : " + toString(eml.getCC()));
+			LOGGER.debug("bcc : " + toString(eml.getBcc()));
+			LOGGER.debug("sender : " + toString(eml.getSender()));
+			LOGGER.debug("from : " + toString(eml.getFrom()));
+			LOGGER.debug("sent on : " + toString(eml.getLocalDate()));
+			LOGGER.debug("to : " + toString(eml.getTo()));
+			LOGGER.debug("reversePath : " + toString(eml.getReversePath()));
+			LOGGER.debug("subject : " + eml.getSubject());
+
+			MhtSaveOptions options = MhtSaveOptions.getDefaultMhtml();
+			options.setMhtFormatOptions(MhtFormatOptions.WriteHeader);
+			options.getFormatTemplates().get_Item("From").replace("From:", "Afzender:");
+			options.getFormatTemplates().get_Item("Sent").replace("Sent:", "Verzonden:");
+			options.getFormatTemplates().get_Item("Subject").replace("Subject:", "Onderwerp:");
+			options.getFormatTemplates().get_Item("Importance").replace("To:", "Aan:");
+			options.getFormatTemplates().get_Item("Cc").replace("Cc:", "Afzender:\t");
+
+			// Overrules the default documentname.
+			result.setDocumentName(ConvertorUtil.createTidyNameWithoutExtension(eml.getSubject()));
+
+			// Save the Message to output stream in MHTML format
+			ByteArrayOutputStream emlStream = new ByteArrayOutputStream();
```

Comment:
Using a byte array here (via ByteArrayOutputStream / ByteArrayInputStream) will cause the entire document to be read in memory. It appears more logical to do this via a temporary file, like for other conversions. Or is it different here?